enhancements/sdk-metrics.md

```diff
@@ -0,0 +1,178 @@
+reviewers:
+  - "@dmesser"
+  - "@joelanford"
+approvers:
+  - "@dmesser" 
+  - "@joelanford"
+creation-date: 2020-05-19
+last-updated: 2020-05-19
+status: implementable
+---
+
+# Operator SDK metrics
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The enhancement is intended to define the bundle resources in which SDK labels can be added, so that they can be read by OLM and included in the current OLM metrics. These metrics will help us in providing a measure to evaluate the reachability of Operator SDK among community operators.
+
+## Motivation
+
+Collecting metrics regarding the usage of Operator SDK among operator authors is important to measure the reachability of the tool. Since OLM already has the pipeline built to obtain metrics on the operators and report it to the RedHat telemetry, we intend to utilize the same to obtain data on the operators using SDK to build bundle images.
+
+### Goals
+
+The goal of this enhancement is to :
+1. List the bundle resources on which SDK stamp will be added.
+2. The format of the stamp which will be added.
+
+### Design details
+
+The current bundle image has the following project structure:
+
+```
+ ...
+  bundle.Dockerfile
+  ...
+  $ tree ./deploy/olm-catalog/test-operator/
+  ./deploy/olm-catalog/test-operator/
+  ├── manifests
+  │   ├── example.com_tests_crd.yaml
+  │   └── test-operator.clusterserviceversion.yaml
+  └── metadata
+      └── annotations.yaml
+```
+The proposed SDK stamps will contain the following data:
+* Label indicating the use of SDK
+* Version of SDK used
+* SDK project layout
+
+SDK stamps would be present in the following locations:
+
+1. bundle.Dockerfile
+
+The dockerfile will include the sdk stamp in the form of `LABEL` which will follow the following format:
+
+```
+LABEL sdk.operatorframework.io.metadata.mediatype = "sdk+v1"
+LABEL sdk.operatorframework.io.metadata.version.v1 = sdk_version
+LABEL sdk.operatorframework.io.metadata.layout.v1 = sdk_project_layout
+```
+
+2. annotations.yaml
+
+Similar to the format followed in the dockerfile, `annotations.yaml` will have the following label:
+
+```
+sdk.operatorframework.io.metadata.mediatype = "sdk+v1"
+sdk.operatorframework.io.metadata.version.v1 = sdk_version
+sdk.operatorframework.io.metadata.layout.v1 = sdk_project_layout
+```
+
+For example, in case of memcached-operator having Kubebuilder layout, the labels can be written as:
+
+```
+LABEL sdk.operatorframework.io.metadata.mediatype = "sdk+v1"
+LABEL sdk.operatorframework.io.metadata.version.v1 = "v0.17.0"
+LABEL sdk.operatorframework.io.metadata.layout.v1 = "go.kubebuilder.io/v2.0.0"
+```
+
+Having labels on `bundle.dockerfile` and `annotations.yaml` will help in collecting metrics regarding the use of SDK from indexed operator bundle images.
```

Comment:
Though the enhancement proposes to add stamps to bundle images (`bundle.dockerfile` and `annotations.yaml`) in addition to CSVs, does OLM have/intend to have metrics which would evaluate bundles to extract any useful data. If not, would this be a new metric specifically added for SDK?